src/coreComponents/linearAlgebra/interfaces/direct/SuiteSparse.hpp

```diff
@@ -20,285 +20,133 @@
-
-  /**
-   * @brief Returns the working rank in the sub-communicator
-   * @return the working rank in the sub-communicator
-   */
-  int subCommWorkingRank() const;
-
-  /**
-   * @brief Sets the communicator
-   * @param[in] comm the MPI communicator
-   */
-  void setComm( MPI_Comm const comm );
+  using Base::ready;
+  using Base::matrix;
 
   /**
-   * @brief Returns the communicator
-   * @return the communicator
+   * @brief Compute the preconditioner from a matrix.
+   * @param mat the matrix to precondition.
    */
-  MPI_Comm getComm() const;
+  virtual void setup( Matrix const & mat ) override;
 
   /**
-   * @brief Sets the subcommunicator
-   * @param[in] subComm the MPI subcommunicator
+   * @brief Apply operator to a vector.
+   * @param src input vector (x)
+   * @param dst output vector (b)
+   *
+   * @warning @p src and @p dst cannot alias the same vector.
    */
-  void setSubComm( MPI_Comm const subComm );
+  virtual void apply( Vector const & src, Vector & dst ) const override;
 
   /**
-   * @brief Returns the subcommunicator
-   * @return the subcommunicator
+   * @brief Apply transpose operator to a vector.
+   * @param src input vector (x)
+   * @param dst output vector (b)
+   *
+   * @warning @p src and @p dst cannot alias the same vector (some implementations may allow this).
    */
-  MPI_Comm getSubComm() const;
+  void applyTranspose( Vector const & src, Vector & dst ) const;
 
   /**
-   * @brief Returns the number of rows
-   * @returns the number of rows
+   * @brief Clean up the preconditioner setup.
    */
-  SSInt numRows() const;
+  virtual void clear() override;
 
   /**
-   * @brief Returns the number of columns
-   * @return the number of columns
+   * @brief Solve the system with a particular rhs
+   * @param [in] rhs system right hand side
+   * @param [out] sol system solution
    */
-  SSInt numCols() const;
+  virtual void solve( Vector const & rhs, Vector & sol ) const override;
 
-  /**
-   * @brief Returns the number of non zeros
-   * @return the number of non zeros
-   */
-  SSInt nonZeros() const;
-
-  /**
-   * @brief Allocate the internal data storage arrays
-   * @param[in] numRows the number of rows
-   * @param[in] numCols the number of columns
-   * @param[in] nonZeros the number of non zeros
-   */
-  void resize( SSInt const numRows, SSInt const numCols, SSInt const nonZeros );
+private:
 
-  /**
-   * @brief Returns the array with the row pointers
-   * @return the array with the row pointers
-   */
-  arrayView1d< SSInt > rowPtr();
+  using Base::m_params;
+  using Base::m_result;
 
   /**
-   * @brief Returns the array with the column indices
-   * @return the array with the column indices
+   * @brief Perform the actual solution.
+   * @param b the rhs vector
+   * @param x the solution vector
+   * @param transpose whether to do a regular or a transpose solve
    */
-  arrayView1d< SSInt > colIndices();
+  void doSolve( Vector const & b, Vector & x, bool transpose ) const;
```

Comment:
I actually think this is too specific... the role of that function is to call the underlying library solve routine, plus gather/scatter stuff, and we don't really care that the solve is done via LU substitution or some other way, especially since this is a private method and not exposed to client code.